openvasd/openvasd.h

```diff
@@ -0,0 +1,300 @@
+                     gchar *);
+
+void
+openvasd_result_free (openvasd_result_t *);
+
+gchar *openvasd_get_result_member_str (openvasd_result_t,
+                                       openvasd_result_member_string_t);
+
+int openvasd_get_result_member_int (openvasd_result_t,
+                                    openvasd_result_member_int_t);
+
+int
+openvasd_parsed_results (openvasd_connector_t *, unsigned long, unsigned long,
+                         GSList **);
+
+openvasd_resp_t
+openvasd_get_scan_status (openvasd_connector_t *);
+
+openvasd_scan_status_t
+openvasd_parsed_scan_status (openvasd_connector_t *);
+
+int
+openvasd_get_scan_progress (openvasd_connector_t *);
+
+openvasd_resp_t
+openvasd_get_health_alive (openvasd_connector_t *);
+
+openvasd_resp_t
+openvasd_get_health_ready (openvasd_connector_t *);
+
+openvasd_resp_t
+openvasd_get_health_started (openvasd_connector_t *);
+
+/* Scanner preferences */
+
+typedef struct openvasd_param openvasd_param_t;
+
+openvasd_resp_t
+openvasd_get_scan_preferences (openvasd_connector_t *);
+
+int
+openvasd_parsed_scans_preferences (openvasd_connector_t *, GSList **);
+
+void
+openvasd_param_free (openvasd_param_t *);
+
+char *
+openvasd_param_id (openvasd_param_t *);
+
+char *
+openvasd_param_name (openvasd_param_t *);
+
+char *
+openvasd_param_desc (openvasd_param_t *);
+
+int
+openvasd_param_mandatory (openvasd_param_t *);
+
+char *
+openvasd_param_type (openvasd_param_t *);
+
+char *
+openvasd_param_default (openvasd_param_t *);
+
+/* Target builder */
+
+typedef struct openvasd_target openvasd_target_t;
+
+typedef struct openvasd_vt_single openvasd_vt_single_t;
+
+typedef struct openvasd_credential openvasd_credential_t;
+
+openvasd_target_t *
+openvasd_target_new (const char *, const char *, const char *, const char *,
+                     int, int);
+
+void
+openvasd_target_set_finished_hosts (openvasd_target_t *, const char *);
+
+void
+openvasd_target_add_alive_test_methods (openvasd_target_t *, gboolean, gboolean,
+                                        gboolean, gboolean, gboolean);
+
+void
+openvasd_target_free (openvasd_target_t *);
+
+openvasd_credential_t *
+openvasd_credential_new (const char *, const char *, const char *);
+
+void
+openvasd_credential_set_auth_data (openvasd_credential_t *, const char *,
+                                   const char *);
+void
+openvasd_credential_free (openvasd_credential_t *);
+
+void
+openvasd_target_add_credential (openvasd_target_t *, openvasd_credential_t *);
+
+openvasd_vt_single_t *
+openvasd_vt_single_new (const char *);
+
+void
+openvasd_vt_single_free (openvasd_vt_single_t *);
+void
+
+openvasd_vt_single_add_value (openvasd_vt_single_t *, const char *,
+                              const char *);
+
+/* Scan config builder */
+gchar *
+openvasd_build_scan_config_json (openvasd_target_t *, GHashTable *, GSList *);
+
+/* Curl multiperform wrapper */
+
+typedef void *curlm_t;
```

Comment:
I think the naming should be prefixed with openvasd_ as well. Otherwise it could happen that another header may redefine `curlm_t`